spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Cell.scala

```diff
@@ -80,7 +80,11 @@ abstract class Cell[T : ClassTag](
    * Please refer to SimpleRNN or LSTM for reference.
    * @return
    */
-  def preTopology: AbstractModule[Activity, Activity, T] = null
+  var preTopology: TensorModule[T]
+
+  private[nn] var includePreTopology: Boolean = false
```

Comment:
I think you also need to handle includePreTopology in the serialization

Reply:
Added